arches/app/utils/index_database.py

```diff
@@ -327,30 +327,41 @@ def index_resources_by_type(
 
     for resource_type in resource_types:
         start = datetime.now()
+        try:
+            uuid.UUID(str(resource_type))
+            graph = models.GraphModel.objects.get(graphid=str(resource_type))
+        except ValueError:
+            try:
+                graph = models.GraphModel.objects.get(slug=str(resource_type))
+            except:
+                logger.warning(
+                    "Unable to resolve resource type %s. Please confirm it is a valid graph ID or slug."
+                    % resource_type
+                )
+                continue
 
-        graph_name = models.GraphModel.objects.get(graphid=str(resource_type)).name
-        logger.info("Indexing resource type '{0}'".format(graph_name))
+        logger.info("Indexing resource type '{0}'".format(graph.name))
 
         if clear_index:
             tq = Query(se=se)
             cards = models.CardModel.objects.filter(
-                graph_id=str(resource_type)
+                graph_id=str(graph.graphid)
             ).select_related("nodegroup")
             for nodegroup in [card.nodegroup for card in cards]:
                 term = Term(field="nodegroupid", term=str(nodegroup.nodegroupid))
```

Comment:
Up to you, but your changes enable making this a lot simpler:
```suggestion
            for card in graph.cardmodel_set.all():
                term = Term(field="nodegroupid", term=str(card.nodegroup_id))
```

Reply:
Good point - wasn't thinking about refactoring anything when I was in there but this makes way more sense.